avm/res/network/trafficmanagerprofile/tests/e2e/defaults/main.test.bicep

```diff
@@ -0,0 +1,47 @@
+targetScope = 'subscription'
+
+metadata name = 'Using only defaults'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-network.trafficmanagerprofiles-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'ntmpmin'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+var resourceName = '${namePrefix}${serviceShort}001'
```

Comment:
Should this logic not be in the resource deployment directly?

Reply:
I believe it was declared as a variable because it's used twice in name and relative name. But, do you think it would be easier to read if the logic was added to both name and relative name and delete the variable?

Reply:
Agreed, made relativeName nullable, so omitting in default and waf test cases and including it in the resource deployment for the max test case. Added a '-rn' on the logic for relativeName in the test case to test the param with a different name than the name param.